src/patient-search/patient-search.component.tsx

```diff
@@ -1,10 +1,96 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import { match } from "react-router";
+import { debounce } from "lodash";
+import styles from "./patient-search.component.css";
+import { performPatientSearch } from "./patient-search.resource";
+import PatientSearchResults from "./patient-search-result/patient-search-result.component";
 
 export default function PatientSearch(props: PatientSearchProps) {
-  return <section>The patient search!</section>;
+  const searchTimeout = 400;
+  const [searchResults, setSearchResults] = useState([]);
+  const [emptyResult, setEmptyResult] = useState(false);
+  const [searchTerm, setSearchTerm] = useState("");
+
+  useEffect(() => {
+    if (searchTerm) {
+      performPatientSearch(searchTerm, "full").then(response => {
+        const { data } = response;
+        setSearchResults(data["results"]);
+        if (data["results"].length === 0) {
+          setEmptyResult(true);
+        } else {
+          setEmptyResult(false);
+        }
+      });
+    } else {
+      setSearchResults([]);
+    }
+  }, [searchTerm]);
+
+  const handleChange = debounce(searchTerm => {
+    setSearchTerm(searchTerm);
+  }, searchTimeout);
+
+  return (
+    <div>
+      <div className={styles.patientSearchHeader}>
+        <button onClick={props.history.goBack} className={styles.iconBtn}>
+          <svg className="omrs-icon" fill="var(--omrs-color-interaction)">
+            <use xlinkHref="#omrs-icon-arrow-back" />
+          </svg>
+        </button>
+        <input
+          className={`omrs-type-title-5 ${styles.patientSearchInput}`}
+          placeholder="Search for patient"
+          aria-label="search"
+          onChange={$event => handleChange($event.target.value)}
+          autoFocus
+        />
+      </div>
+      <div className={styles.searchResults}>
+        {searchResults.length ? (
+          <div>
+            <div className={styles.resultsCount}>
+              <p>
+                <span className={styles.resultsText}>Results:</span>{" "}
+                {searchResults.length}{" "}
+              </p>
+            </div>
+            <PatientSearchResults patients={searchResults} />
+          </div>
+        ) : null}
+      </div>
+      {!searchResults.length && !emptyResult ? (
```

Comment:
nit: if you don't really want to have an `else` branch, you can use `&&` instead of ternaries:

```suggestion
      {!searchResults.length && !emptyResult && (
```